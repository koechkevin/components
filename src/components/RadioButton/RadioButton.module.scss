@import '../../styles/_global.scss';

.radioButton {
  :global(.ant-radio-wrapper) {
    display: table;
  }

  :global(.ant-radio) {
    display: table-cell;
    vertical-align: middle;
    width: 32px;
    height: 32px;
    padding: 8px;

    &:hover {
      :global(.ant-radio-inner) {
        border-color: $color-primary !important;
      }
    }
  }

  :global(span.ant-radio + *) {
    display: table-cell;
    vertical-align: middle;
  }

  :global(.ant-radio.ant-radio-checked .ant-radio-inner) {
    background-color: $color-primary;
    border-color: $color-primary;

    &:after {
      background-color: $color-white;
      transform: scale(0.8);
    }
  }

  :global(.ant-radio-disabled) {
    &:hover {
      :global(.ant-radio-inner) {
        border-color: $color-light-gray !important;
      }
    }
  }

  :global(.ant-radio-disabled.ant-radio-checked) {
    cursor: default;

    &:hover {
      :global(.ant-radio-inner) {
        border-color: transparent !important;
      }
    }
  }
}
