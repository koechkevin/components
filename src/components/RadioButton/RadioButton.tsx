import { Radio as AntRadio } from 'antd';
import { RadioProps } from 'antd/lib/radio';
import React, { FC, PureComponent } from 'react';

import styles from './RadioButton.module.scss';

interface Props extends RadioProps {
  checked?: boolean;
  label?: string;
  onSelect?: () => void;
}

export const Radio: FC<RadioProps> = (props) => {
  const { className, ...restProps } = props;
  const classNames = [styles.radioButton, className].join(' ');

  return <AntRadio className={classNames} {...restProps} />;
};

class RadioButton extends PureComponent<Props> {
  public render = () => <Radio {...this.props} />;
}

export default RadioButton;
