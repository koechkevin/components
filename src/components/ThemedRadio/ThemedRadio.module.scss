@import '../../styles/_global.scss';

.radio {
  cursor: pointer;
  border-radius: 4px;
  width: 100%;
  padding: 8px;
  background-color: $color-basic-light-gray;
  margin-bottom: 8px;

  &:global(.ant-radio-wrapper) {
    padding: 8px;
    margin-bottom: 8px;
  }

  &.primary,
  &.blue {
    color: $color-primary;
    background-color: $color-primary-light;
    :global {
      .ant-typography {
        color: $color-primary;
        background-color: $color-primary-light;
      }
      .ant-radio-inner {
        background-color: $color-primary;
        border-color: $color-primary;
        &:after {
          background-color: $color-white;
          transform: scale(0.8);
        }
      }
    }
  }

  &.green {
    color: $color-radio-green;
    background-color: $color-radio-green-light;
    :global {
      .ant-typography {
        color: $color-radio-green;
        background-color: $color-radio-green-light;
      }

      .ant-radio-inner {
        background-color: $color-radio-green;
        border-color: $color-radio-green;
        &:after {
          background-color: $color-white;
          transform: scale(0.8);
        }
      }
    }
  }

  &.gold {
    color: $color-gold-radio;
    background-color: $color-gold-radio-light;
    :global {
      .ant-typography {
        color: $color-gold-radio;
        background-color: $color-gold-radio-light;
      }

      .ant-radio-inner {
        background-color: $color-gold-radio;
        border-color: $color-gold-radio;
        &:after {
          background-color: $color-white;
          transform: scale(0.8);
        }
      }
    }
  }

  &.purple {
    color: $color-purple-radio;
    background-color: $color-purple-radio-light;
    :global {
      .ant-typography {
        color: $color-purple-radio;
        background-color: $color-purple-radio-light;
      }

      .ant-radio-inner {
        background-color: $color-purple-radio;
        border-color: $color-purple-radio;
        &:after {
          background-color: $color-white;
          transform: scale(0.8);
        }
      }
    }
  }
}
