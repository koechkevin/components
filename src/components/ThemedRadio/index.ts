export { default as ThemedRadio } from './ThemedRadio'
