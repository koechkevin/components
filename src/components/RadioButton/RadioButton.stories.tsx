import { storiesOf } from '@storybook/react';
import { Col, Row, Typography } from 'antd';

import React from 'react';

import RadioButton from './RadioButton';

const stories = storiesOf('Radio Button', module);

stories.addDecorator((story) => <Row>{story()}</Row>);

stories.add('default', () => {
  return (
    <Row type="flex" style={{ padding: '5%' }} justify="center" gutter={12}>
      <Col span={24}>
        <Typography.Text strong>Normal button</Typography.Text>
        <br />
        <RadioButton>Yes</RadioButton>
      </Col>
      <Col span={24}>
        <Typography.Text strong>Disabled Button</Typography.Text>
        <br />
        <RadioButton disabled>Disabled</RadioButton>
      </Col>
    </Row>
  );
});
