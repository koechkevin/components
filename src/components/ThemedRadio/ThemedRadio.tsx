import { RadioProps } from 'antd/lib/radio';
import React, { FC } from 'react';
import RadioButton from '../RadioButton/RadioButton';

import { Text } from '../Typography';
import styles from './ThemedRadio.module.scss';

interface Props extends RadioProps {
  checked?: boolean;
  label?: string;
  theme: string;
  onSelect?: () => void;
}

const ThemedRadio: FC<Props> = (props) => {
  const { checked, label, theme } = props;

  return (
    <RadioButton {...props} className={[styles.radio, checked ? styles[theme] : ''].join(' ')}>
      <Text>{label}</Text>
    </RadioButton>
  );
};

export default ThemedRadio;
