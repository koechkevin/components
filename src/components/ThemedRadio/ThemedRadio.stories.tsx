import { boolean, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { Col, Row, Typography } from 'antd';

import React from 'react';

import ThemedRadio from './ThemedRadio';

const stories = storiesOf('Radio Button', module);

stories.addDecorator((story) => <Row>{story()}</Row>);

stories.add('Themed Radio', () => {
  return (
    <Row type="flex" style={{ padding: '5%'}} justify="center" gutter={12}>
      <Col span={8}>
        <Typography.Text strong>Blue or Primary Theme</Typography.Text>
        <ThemedRadio checked={true} label="Admin" theme="blue" onSelect={() => {}} />
      </Col>
      <Col span={8}>
        <Typography.Text strong>Green Theme</Typography.Text>
        <ThemedRadio checked={true} label="Recruiter" theme="green" onSelect={() => {}} />
      </Col>
      <Col span={8}>
        <Typography.Text strong>Gold Theme</Typography.Text>
        <ThemedRadio checked={true} label="Owner" theme="gold" onSelect={() => {}} />
      </Col>
      <Col span={8}>
        <Typography.Text strong>Purple Theme</Typography.Text>
        <ThemedRadio checked={true} label="Billing" theme="purple" onSelect={() => {}} />
      </Col>

      <Col span={8}>
        <Typography.Text strong>Toggle theme and Checked using knobs</Typography.Text>
        <ThemedRadio
          checked={boolean('checked', true)}
          label="Owner"
          theme={select('theme', ['purple', 'green', 'blue', 'gold'], 'green')}
          onSelect={() => {}}
        />
      </Col>
    </Row>
  );
});
