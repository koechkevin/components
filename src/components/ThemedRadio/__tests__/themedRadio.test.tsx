import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import RadioButton from '../../RadioButton/RadioButton';
import ThemedRadio from '../ThemedRadio';

describe('themed radio button', () => {
  it('renders checked button without crashing', () => {
    const wrapper: ReactWrapper = mount(
      <ThemedRadio checked={true} label="Owner" theme={'blue'} onSelect={jest.fn()} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders unchecked button without crashing', () => {
    const wrapper: ReactWrapper = mount(
      <ThemedRadio checked={false} label="Owner" theme={'primary'} onSelect={jest.fn()} />,
    );
    expect(wrapper.find(RadioButton)).toHaveLength(1);
  });
});
