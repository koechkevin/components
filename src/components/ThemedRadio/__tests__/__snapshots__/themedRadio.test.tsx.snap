// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themed radio button renders checked button without crashing 1`] = `
<ThemedRadio
  checked={true}
  label="Owner"
  onSelect={[MockFunction]}
  theme="blue"
>
  <RadioButton
    checked={true}
    className="radio blue"
    label="Owner"
    onSelect={[MockFunction]}
    theme="blue"
  >
    <Radio
      checked={true}
      className="radio blue"
      label="Owner"
      onSelect={[MockFunction]}
      theme="blue"
    >
      <Radio
        checked={true}
        className="radioButton radio blue"
        label="Owner"
        onSelect={[MockFunction]}
        theme="blue"
        type="radio"
      >
        <label
          className="radioButton radio blue ant-radio-wrapper ant-radio-wrapper-checked"
        >
          <Checkbox
            checked={true}
            className=""
            defaultChecked={false}
            label="Owner"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onSelect={[MockFunction]}
            prefixCls="ant-radio"
            style={Object {}}
            theme="blue"
            type="radio"
          >
            <span
              className="ant-radio ant-radio-checked"
              style={Object {}}
            >
              <input
                checked={true}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
          </Checkbox>
          <span>
            <Text>
              <Text
                className="text "
              >
                <withConfigConsumer(Base)
                  className="text "
                  component="span"
                  ellipsis={false}
                >
                  <Base
                    className="text "
                    component="span"
                    ellipsis={false}
                    getPrefixCls={[Function]}
                    prefixCls="ant-typography"
                    renderEmpty={[Function]}
                  >
                    <LocaleReceiver
                      componentName="Text"
                    >
                      <ResizeObserver
                        disabled={true}
                        onResize={[Function]}
                      >
                        <Typography
                          aria-label={null}
                          className="text "
                          component="span"
                          style={
                            Object {
                              "WebkitLineClamp": null,
                            }
                          }
                        >
                          <span
                            aria-label={null}
                            className="ant-typography text "
                            style={
                              Object {
                                "WebkitLineClamp": null,
                              }
                            }
                          >
                            Owner
                          </span>
                        </Typography>
                      </ResizeObserver>
                    </LocaleReceiver>
                  </Base>
                </withConfigConsumer(Base)>
              </Text>
            </Text>
          </span>
        </label>
      </Radio>
    </Radio>
  </RadioButton>
</ThemedRadio>
`;
